trunk/SUAVE/Methods/Aerodynamics/Airfoil_Panel_Method/thwaites_method.py

```diff
@@ -0,0 +1,292 @@
+            Re_x[0]     = 1E-5
+            
+            # Store results 
+            X_T[:,a_i,re_i]          = x
+            THETA_T[:,a_i,re_i]      = theta
+            DELTA_STAR_T[:,a_i,re_i] = del_star
+            H_T[:,a_i,re_i]          = H
+            CF_T[:,a_i,re_i]         = cf
+            RE_THETA_T[:,a_i,re_i]   = Re_theta
+            RE_X_T[:,a_i,re_i]       = Re_x
+            DELTA_T[:,a_i,re_i]      = delta 
+    
+    RESULTS = Data(
+        X_T          = X_T,      
+        THETA_T      = THETA_T,   
+        DELTA_STAR_T = DELTA_STAR_T,
+        H_T          = H_T,       
+        CF_T         = CF_T,      
+        RE_THETA_T   = RE_THETA_T,   
+        RE_X_T       = RE_X_T,    
+        DELTA_T      = DELTA_T,  
+    )    
+    
+    return RESULTS
+    
+def getH(lambda_val ): 
+    """ Computes the shape factor, H
+
+    Assumptions:
+    None
+
+    Source:
+    None
+
+    Inputs: 
+    lamdda_val  - thwaites separation criteria 
+
+    Outputs:  
+    H           - shape factor
+
+    Properties Used:
+    N/A
+    """       
+    H       = 0.0731/(0.14 + lambda_val ) + 2.088 
+    idx1    = (lambda_val>0.0)  
+    H[idx1] = 2.61 - 3.75*lambda_val[idx1]  + 5.24*lambda_val[idx1]**2   
+    return H 
+    
+def odefcn(y,x, nu,x_i,Ve_i):
+    """ Computes bounday layer functions using SciPy ODE solver 
+
+    Assumptions:
+    None
+
+    Source:
+    None
+
+    Inputs: 
+    y           - initial conditions of functions 
+    x           - new x values at which to solve ODE
+    nu          - kinematic viscosity 
+    x_i         - intial array of x values 
+    Ve_i        - intial boundary layer velocity 
+    
+    Outputs:  
+    dydx        - expression for the momentum thickness and velocity (theta**2/Ve**6)
+
+    Properties Used:
+    N/A 
+    """        
+    dydx = 0.45*getVe(x,x_i,Ve_i)**5*nu
+    return dydx 
+    
+def getVe(x,x_i,Ve_i):
+    """ Interpolates the bounday layer velocity over a new dimension of x 
+
+    Assumptions:
+    None
+
+    Source:
+    None
+
+    Inputs: 
+    x         - new x dimension
+    x_i       - old x dimension 
+    Ve_i      - old boundary layer velocity values  
+    
+    Outputs:  
+    Ve        - new boundary layer velocity values 
+
+    Properties Used:
+    N/A 
+    """
+    Ve_func = interp1d(x_i,Ve_i, axis=0,fill_value = "extrapolate")
+    Ve      = Ve_func(x)
+    return Ve 
+
+
+def getcos(x,x_i,cos_i):
+    """ Interpolates the bounday layer velocity over a new dimension of x 
+
+    Assumptions:
+    None
+
+    Source:
+    None
+
+    Inputs: 
+    x         - new x dimension
+    x_i       - old x dimension 
+    cos_i      - old boundary layer velocity values  
+    
+    Outputs:  
+    cos        - new boundary layer velocity values 
+
+    Properties Used:
+    N/A 
+    """
+    cos_func = interp1d(x_i,cos_i, axis=0,fill_value = "extrapolate")
+    cos_t     = cos_func(x)
+    return cos_t
+
+def getdVe(x,x_i,dVe_i):
```

Comment:
this is only used once; maybe remove the extra function for it

Reply:
It might useful to have explicit functions in the future for debugging so I will leave it in for now.